lms/djangoapps/grades/api/views.py

```diff
@@ -128,28 +137,48 @@ def get(self, request, course_id):
         Return:
             A JSON serialized representation of the requesting user's current grade status.
         """
-        username = request.GET.get('username')
-
-        # only the student can access her own grade status info
-        if request.user.username != username:
-            log.info(
-                'User %s tried to access the grade for user %s.',
-                request.user.username,
-                username
-            )
-            return self.make_error_response(
-                status_code=status.HTTP_404_NOT_FOUND,
-                developer_message='The user requested does not match the logged in user.',
-                error_code='user_mismatch'
-            )
 
+        # Use the request user's if none provided.
```

Comment:
After reading more of the code I think it would be useful to describe the cases when there would be a username in the request, and when there would not so we would use the request.user username instead.